src/trading/TradingEscapeHatch.sol

```diff
@@ -0,0 +1,96 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/trading/ITradingEscapeHatch.sol';
+import 'ROOT/trading/ICash.sol';
+import 'ROOT/trading/IOrders.sol';
+import 'ROOT/trading/IShareToken.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+
+contract TradingEscapeHatch is DelegationTarget, ITradingEscapeHatch {
+    using SafeMathUint256 for uint256;
+
+    //      market             outcome  value
+    mapping(address => mapping(uint8 => uint256)) private frozenShareValues;
+
+    function claimSharesInUpdate(IMarket _market) public onlyInBadTimes returns(bool) {
+        require(address(_market) != address(0));
+
+        uint8 _numOutcomes = _market.getNumberOfOutcomes();
+        ICash _marketCurrency = _market.getDenominationToken();
+
+        for (uint8 _outcome = 0; _outcome < _market.getNumberOfOutcomes(); ++_outcome) {
+            IShareToken _shareToken = _market.getShareToken(_outcome);
+            uint256 _sharesOwned = _shareToken.balanceOf(msg.sender);
+            if (_sharesOwned > 0) {
+                uint256 _frozenShareValue = getFrozenShareValue(_market, _outcome);
+                _shareToken.destroyShares(msg.sender, _sharesOwned);
+                uint256 _amountToTransfer = _sharesOwned.fxpMul(_frozenShareValue, 1 ether);
+                _marketCurrency.transferFrom(_market, msg.sender, _amountToTransfer);
```

Comment:
If I change `_amountToTransfer` to `0`, there is no exception. If I change it to `1`, there's an exception.

Reply:
In Market.sol the method 'approveSpenders' needs to update the capitalization for this contract.